pallet/src/lib.rs

```diff
@@ -152,9 +168,60 @@ pub mod pallet {
     }
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-        fn on_idle(_n: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
-            // TODO: Add maintainance for storage MultisigStorage (remove old requests).
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T>
+    where
+        BalanceOf<T>: From<u128> + Into<u128>,
+    {
+        fn on_idle(block_number: BlockNumberFor<T>, mut remaining_weight: Weight) -> Weight {
+            let len = MultisigStorage::<T>::iter_keys().count() as u64;
+            let lifetime = T::MaxLifetimeRequests::get();
+            // Abort if storage is empty or the allowed lifetime is longer than the blockchain's
+            // existence (otherwise, an integer underflow can occur).
+            if len == 0 || block_number < lifetime {
+                return remaining_weight - T::DbWeight::get().reads(1);
+            } else {
+                remaining_weight -= T::DbWeight::get().reads_writes(3, 1);
+            }
```

Comment:
`else` block is not needed due to unconditional return in the first block
```suggestion
            if len == 0 || block_number < lifetime {
                return remaining_weight - T::DbWeight::get().reads(1);
            }
            
            remaining_weight -= T::DbWeight::get().reads_writes(3, 1);
```

Reply:
Done